Cmdr/Shared/Command.lua

```diff
@@ -18,6 +18,14 @@ function Command.new (options)
 		Object = options.CommandObject; -- The command object (definition)
 		Group = options.CommandObject.Group; -- The group this command is in
 		State = {}; -- A table which will hold any custom command state information
+		HasImplementation = function() --Returns true if the command has an implementation on the caller's machine.
+			if RunService:IsClient() then
+				return options.CommandObject.ClientRun and true or false
+			elseif RunService:IsServer() then
+				--Assuming the server implementation of the command is the Run function of CommandObject.
+				return options.CommandObject.Run and true or false
+			end
+		end
```

Comment:
This should go on the Command prototype. Creating a new function every time isn't great for memory